### PR TITLE
Add docker compose and bake

### PR DESCRIPTION
**Pull request by Mandur**
# PR for issue #585

## What is being addressed

CI should build containers for the lora network server.

## How is this addressed

We build the container for ARM and AMD64 as part of a new job in the CI.
I also built during the tests a docker-compose.yaml that could be useful for a release pipeline. Therefore, I also put it on the pr. (Docker-compose build will build all the containers of the solution, but that doesnt' work on GH action)

### CODE REVIEW

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/Dockerfile.arm32v7 [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
 
-RUN dotnet publish -c Release -o out
+RUN dotnet publish "LoRaWanNetworkSrvModule.csproj" -c Release -o out --no-restore
 
-FROM mcr.microsoft.com/dotnet/aspnet:3.1-buster-slim-arm32v7
+FROM mcr.microsoft.com/dotnet/aspnet:3.1-bullseye-slim-arm32v7
 WORKDIR /app
 COPY --from=build-env /build/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWanNetworkSrvModule/out/* ./
+
+RUN useradd -ms /bin/bash moduleuser
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/Dockerfile.arm32v7:31]: What is this for?

**Review by danigian — APPROVED** [review]

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/Dockerfile.arm32v7 (hunk 1 above)
Comment (Mandur) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/Dockerfile.arm32v7:31]: I don't know thought it was already there. Most likely a spillover from another I will fix thanks for the catch

Comment (Mandur) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/Dockerfile.arm32v7:31]: Interestingly this is a spillover from the amd64 dockerfiles. I don't think it's needed, and I would remove this everywhere. I found some ref, I don't think it's needed. WDYT?
https://docs.microsoft.com/en-us/azure/iot-edge/tutorial-machine-learning-edge-06-custom-modules?view=iotedge-2020-11#add-directory-to-the-module

Comment (danigian) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/Dockerfile.arm32v7:31]: As discussed, would make sense to keep it for LNS module, but not for BasicStation one imho

**Pull request merged**